rpc(delegation): Query stakers for operator

TimmyExogenous opened the pull request:
## Description

This PR introduces a command to query the delegated stakers for a specific operator and asset. Additionally, it refines the RPC based on the following suggestion: 
https://github.com/ExocoreNetwork/exocore/pull/270/files#r1891143183

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new CLI command for querying delegated stakers by operator.

- **Bug Fixes**
	- Standardized naming conventions for `StakerID` and `AssetID`, improving consistency across various components.

- **Documentation**
	- Updated field names in API documentation to reflect changes from `StakerID` to `StakerId` and `AssetID` to `AssetId`.

- **Tests**
	- Adjusted test cases to align with updated field names, ensuring validation logic remains intact.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

TimmyExogenous committed:
introduces a command to query the delegated stakers and refine the RPC

MaxMustermann2 commented on app/ethtest_helper.go:
```diff
@@ -225,7 +225,7 @@ func genesisStateWithValSet(codec codec.Codec, genesisState simapp.GenesisState,
 		{
 			Operator: operator.String(),
-			StakerID: stakerID,
+			StakerId: stakerID,
```
To avoid the Lint errors from Golang CI, I changed the name of such fields to Staker. I don't know if they're still triggered.

TimmyExogenous replied:
Yeah, I remember this change. I renamed it to `stakerId` because the proto-generated files use this naming convention. It seems the lint error from Golang CI wasn't triggered, so it should be fine to use `stakerId` now.

MaxMustermann2 commented on x/delegation/client/cli/query.go:
```diff
@@ -277,3 +279,40 @@ func QueryAssociatedStakersByOperator() *cobra.Command {
+			if err != nil {
+				return err
+			}
+			fmt.Println("the number of stakers is:", len(res.Stakers))
```
Remove the debug

TimmyExogenous replied:
It’s not debug information; it’s used to display the number of stakers when running this command

MaxMustermann2 replied:
It'll cause trouble when using `--output json`. Maybe include it in the result instead.

TimmyExogenous replied:
Thanks, it will be fixed in the next commit.

TimmyExogenous committed:
refine the QueryDelegatedStakersByOperator http endpoint
Remove the printing of the staker number and add the staker count to the response.

TimmyExogenous commented on x/delegation/types/genesis.go:
```diff
@@ -197,7 +197,7 @@ func (gs GenesisState) ValidateStakerList() error {
 func (gs GenesisState) ValidateUndelegations() error {
 	validationFunc := func(_ int, undelegation UndelegationRecord) error {
-		err := ValidateIDAndOperator(undelegation.StakerID, undelegation.AssetID, undelegation.OperatorAddr)
+		err := ValidateIDAndOperator(undelegation.StakerId, undelegation.AssetId, undelegation.OperatorAddr)
```
We only use `AssetId` in the code for proto-generation to maintain compatibility with GRPC. I don't think it is necessary to change all `AssetID` to `AssetId`, because it will introduce too many changes to the code.

bwhour reviewed (approved):
LGTM.

TimmyExogenous commented on x/delegation/keeper/grpc_query.go:
```diff
@@ -68,3 +68,15 @@ func (k Keeper) QueryAssociatedStakersByOperator(ctx context.Context, req *deleg
+		return nil, err
+	}
+	return &delegationtype.QueryDelegatedStakersByOperatorResponse{
+		Count:   uint32(len(stakers.Stakers)),
```
thanks, I will use `uint64` to fix this issue

TimmyExogenous commented on proto/exocore/delegation/v1/query.proto:
```diff
@@ -126,49 +126,71 @@ message QueryAssociatedStakersByOperatorResponse {
+  // QueryDelegatedStakersByOperator queries the delegated stakers for the specified operator
+  rpc QueryDelegatedStakersByOperator(QueryDelegatedStakersByOperatorReq) returns (QueryDelegatedStakersByOperatorResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
+    option (google.api.http).get = "/exocore/delegation/v1/delegated_stakers/{operator}";
```
thanks, it will be fixed in the next commit.
